src/main/groovy/com/moowork/gradle/node/npm/NpmInstallTask.groovy

```diff
@@ -24,25 +24,54 @@ class NpmInstallTask
         this.project.afterEvaluate {
             def ext = this.project.extensions.getByType(NodeExtension)
             setNpmCommand( ext.getNpmInstallCommand() )
+        }
+    }
+
+    @InputFile
+    @Optional
+    @PathSensitive(PathSensitivity.RELATIVE)
+    protected getPackageJsonFile()
+    {
+        def file = new File( (File) this.project.node.nodeModulesDir, 'package.json' )
+        return file.exists() ? file : null
+    }
+
+    @InputFile
+    @Optional
+    @PathSensitive(PathSensitivity.RELATIVE)
+    protected getNpmShrinkwrap()
+    {
+        def file = new File( (File) this.project.node.nodeModulesDir, 'npm-shrinkwrap.json' )
+        return file.exists() ? file : null
+    }
 
-            getInputs().file( new File( (File) this.project.node.nodeModulesDir, 'package.json' ) )
-                    .withPathSensitivity(PathSensitivity.RELATIVE)
-            getInputs().file( new File( (File) this.project.node.nodeModulesDir, 'npm-shrinkwrap.json' ) )
-                    .withPathSensitivity(PathSensitivity.RELATIVE)
-                    .optional()
-            getOutputs().dir( new File( (File) this.project.node.nodeModulesDir, 'node_modules' ) )
-
-            def lockFile = new File(this.project.extensions.getByType(NodeExtension).nodeModulesDir, 'package-lock.json')
-            if (npmCommand[0] == "ci") {
-                getInputs().file(lockFile.exists() ? lockFile : null)
-                        .withPathSensitivity(PathSensitivity.RELATIVE)
-                        .optional()
-            } else if (npmCommand[0] == "install") {
-                getOutputs().file(lockFile)
-            }
+    @InputFile
+    @Optional
+    @PathSensitive(PathSensitivity.RELATIVE)
+    protected getPackageLockFileAsInput()
+    {
+        def lockFile = new File(this.project.extensions.getByType(NodeExtension).nodeModulesDir, 'package-lock.json')
+        if (npmCommand[0] == "ci") {
+            return lockFile.exists() ? lockFile : null
+        }
+
+        return null
+    }
+
+    @OutputFile
+    @Optional
+    @PathSensitive(PathSensitivity.RELATIVE)
+    protected getPackageLockFileAsOutput()
+    {
+        if (npmCommand[0] == "install") {
+            def file = new File(this.project.extensions.getByType(NodeExtension).nodeModulesDir, 'package-lock.json')
```

Comment:
am not sure about this. isnt the package-lock.json an output only if it does not exist?

Reply:
Judging by https://github.com/npm/npm/issues/17979#issuecomment-332701215 it's created if it does not exist but it's also updated if you make changes to `package.json`

Reply:
ok